## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 496233`
Commit: [`b5e6ad21f133d496029eddad02ed2a2738d1d579`](https://github.com/NixOS/nixpkgs/commit/b5e6ad21f133d496029eddad02ed2a2738d1d579) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/b5e6ad21f133d496029eddad02ed2a2738d1d579..pull/496233/head))
Merge: [`a964d733b03dce2e78147272603a81c26de64e22`](https://github.com/NixOS/nixpkgs/commit/a964d733b03dce2e78147272603a81c26de64e22)

Logs: https://github.com/airone01/nixpkgs-review-gha/actions/runs/22649358334


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>searchix</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>searchix</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>searchix</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>searchix</summary>
<pre>Running phase: unpackPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/iaar1dmxh5pd36kxc6cf19spn7z8i4mq-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
Building subPackage ./cmd/searchix-web
buildPhase completed in 1 minutes 28 seconds
Running phase: installPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;installPhase&quot; }
panic: open /etc/protocols: operation not permitted

goroutine 1 [running]:
modernc.org/libc/honnef.co/go/netdb.init.0()
	modernc.org/libc@v1.65.10/honnef.co/go/netdb/netdb.go:700 +0x713</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>searchix</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>searchix</summary>
<pre>Running phase: unpackPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/iaar1dmxh5pd36kxc6cf19spn7z8i4mq-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
Building subPackage ./cmd/searchix-web
Running phase: installPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;installPhase&quot; }
panic: open /etc/protocols: operation not permitted

goroutine 1 [running]:
modernc.org/libc/honnef.co/go/netdb.init.0()
	modernc.org/libc@v1.65.10/honnef.co/go/netdb/netdb.go:700 +0x5b8</pre>
</details>
</details>

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -895,6 +895,12 @@
     githubId = 37664775;
     name = "Yuto Oguchi";
   };
+  airone01 = {
+    name = "Erwann Lagouche";
+    github = "airone01";
+    githubId = 21955960;
+    email = "airone01@proton.me";
+  };
   airrnot = {
     name = "airRnot";
     github = "airRnot1106";

--- a/pkgs/by-name/se/searchix/package.nix
+++ b/pkgs/by-name/se/searchix/package.nix
@@ -1,0 +1,74 @@
+{
+  lib,
+  stdenv,
+  buildGoModule,
+  fetchFromGitea,
+  fetchFromGitHub,
+  versionCheckHook,
+  nix-update-script,
+}:
+
+let
+  simpleCss = fetchFromGitHub {
+    owner = "kevquirk";
+    repo = "simple.css";
+    rev = "ba4af949057d489331759e0118de596222e0f5b7";
+    hash = "sha256-rihjNW1gf0k7DI8x+vaFUR4ehI3gXDV9zWV3DGSg4y8=";
+  };
+in
+
+buildGoModule (finalAttrs: {
+  pname = "searchix";
+  version = "0.4.4";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "alinnow";
+    repo = "searchix";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-R8/iR8QoeUwNQCFkr+JtoPq/iKkJRIe8dsbvGfFqinE=";
+  };
+
+  vendorHash = "sha256-yfcQgy4cQFRvtsyLHLojnJaWhle1ZR3unmaFQj8ljuw=";
+
+  subPackages = [ "cmd/searchix-web" ];
+
+  tags = [ "embed" ];
+
+  ldflags = [
+    "-s"
+    "-X=alin.ovh/searchix/internal/config.Version=${finalAttrs.version}"
+  ];
+
+  preBuild = ''
+    rm -f frontend/static/base.css
+    cp ${simpleCss}/simple.css frontend/static/base.css
+  '';
+
+  postInstall = lib.optionalString stdenv.hostPlatform.isLinux ''
+    $out/bin/searchix-web generate-error-page --outdir $out/share/searchix/
+  '';
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgramArg = "version";
+  doInstallCheck = true;
+
+  passthru = {
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    description = "Search tool for options and packages in the NixOS ecosystem";
+    homepage = "https://searchix.ovh/";
+    downloadPage = "https://codeberg.org/alinnow/searchix";
+    changelog = "https://codeberg.org/alinnow/searchix/src/tag/${finalAttrs.src.tag}/CHANGELOG.md";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [
+      airone01
+      BatteredBunny
+    ];
+    mainProgram = "searchix-web";
+  };
+})

